src/KubernetesClient/Watcher.cs

```diff
@@ -177,13 +183,119 @@ public static class WatcherExt
         /// <param name="onClosed">
         /// The action to invoke when the server closes the connection.
         /// </param>
+        /// <param name="cancellationToken">A token that can be used to cancel the read</param>
         /// <returns>a watch object</returns>
         public static Watcher<T> Watch<T>(this HttpOperationResponse<T> response,
             Action<WatchEventType, T> onEvent,
             Action<Exception> onError = null,
-            Action onClosed = null)
+            Action onClosed = null,
+            CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return Watch((HttpOperationResponse)response, onEvent, onError, onClosed, cancellationToken);
+        }
+
+        /// <summary>
+        /// watch the call to api server with watch=true asynchronously
+        /// </summary>
+        /// <typeparam name="T">type of the event object</typeparam>
+        /// <param name="response">the api response</param>
+        /// <param name="onEvent">a callback when any event raised from api server</param>
+        /// <param name="onError">a callbak when any exception was caught during watching</param>
+        /// <param name="onClosed">
+        /// The action to invoke when the server closes the connection.
+        /// </param>
+        /// <param name="cancellationToken">A token that can be used to cancel the watch</param>
+        /// <returns>a watch object</returns>
+        public static async Task WatchAsync<T>(this HttpOperationResponse response,
+            CancellationToken cancellationToken,
```

Comment:
By convention, CancellationToken should be the last argument.

Reply:
I have moved cancellationToken to be the last argument and defaulted it to default(CancellationToken) as is typical for other .NET core apis.